### ISSUE TITLE
Error caused by invalid language

### ISSUE THREAD

**Issue by webunraveling** [JWDeveloper/JWManagement#398]
**Describe the bug**
When you try to enter publisher availability, you receive a server error message when you try to save the changes.

**To Reproduce**
1. Login and click on a project
2. Click on "Publishers"
3. Click on a publisher who's language value is set to `publisher.entity.profile.languageValues.en`
4. Under availability, click on a day.
5. Click the "Add new timeslot" button
6. Click "Save"
7. An error message pops up saying "SERVER ERROR".

**Additional context**
When a user's language is set to a valid value, the error does not occur.

## Suggested fix?
Sorry I'm not familiar enough with the code yet to know where to look or how exactly other things will be effected. I was experimenting with `imports/framework/Constants/Languages.js` but that didn't seem to make any difference. I haven't been able to track down the bug :angry: 

Any new users will have their language set to English (US) by default. But older users still have a DB value of `en` set in their profile. Would it be worth adding `en` back as an option? Or is there a safe way that we can check their profile setting and change it from `en` to `en-US`? That actually happens automatically when I start `meteor` in my test environment. Would that happen on the staging server?

**Comment by geimist** [comment]
I have a similar problem. A sister can't change her profile language from english to german. When I do this in adminpanel, I get the error: "There was an error with another field: undefined"
When I try to modify another field (congregation), I get the error: "There was an error with another field: en is not an allowed value"

**Comment by webunraveling** [comment]
By the way, it seems that the language is already trying to be reset from `en` to `en-US` in  the develop branch. I don't know if this is active in production right now. See: https://github.com/JWDeveloper/JWManagement/blob/develop/client/views/pages/publisherActions/publisherActions.coffee#L110-L112

**Comment by MarvinZeising** [comment]
Thanks for creating the issue for this.
I just added a little quickfix for this in 1ab7263.
It's being deployed right now. Hope it helps with the problems.

**Comment by webunraveling** [comment]
Yes! I see it was just deployed. Looks like 1ab7263 fixes both issues mentioned here.

**Comment by geimist** [comment]
Unfortunately, the problem persists here. In the admin panel, the profile of the sister is set to _"English (US)"_ by your change, but still cannot be changed (_"There was an error with another field: undefined"_). 

In the profile of the sister the language is _"language.entity.languageValues.en"_. Nothing is checked in the language selection list. Changes are discarded without error message.

<img width="612" alt="Bildschirmfoto 2019-07-04 um 12 06 08" src="https://user-images.githubusercontent.com/29315520/60658774-28c5bd80-9e54-11e9-8674-f2152ba34b6c.png">

**Comment by MarvinZeising** [comment]
Oh right, we could add the language GUI fix here, as well.
But I think that wouldn't fix your problem.
Can you try to change the congregation field again, please?

**Comment by geimist** [comment]
Hallo Marvin,
ich schreibe dir gleich mal auf deutsch ;-)

Seit deinem Fix kann die Schwester ihren Versammlungsnamen selbst ändern; der Admin erhält die obige Fehlermeldung.

Nach einigen Versuchen wird die Website für die Schwester in deutsch ausgegeben, als Sprache steht aber bei ihr immer noch _"language.entity.languageValues.en"_ und im Adminpanel steht _"English (US)"_ 
(Die Schwester hatte mir gesagt, dass die Umstellung schon manchmal nach mehreren Versuchen temporär funktionierte, aber dann immer wieder auf Englisch zurückfällt. Ist aus meiner Sicht auch logisch, wenn ich als Admin in ihrem Profil noch _"English (US)"_ stehen sehe).

**Comment by webunraveling** [comment]
Was this fixed with 1ab7263?

**Comment by geimist** [comment]
The error still exists

**Comment by MarvinZeising** [comment]
Also die Schwester hat über den Link Sprache

![image](https://user-images.githubusercontent.com/12305625/61885826-d03b8a80-aefe-11e9-9b9b-9271e28b686b.png)

ihre Sprache auf Deutsch

![image](https://user-images.githubusercontent.com/12305625/61885847-dc274c80-aefe-11e9-9c6f-ea00029fce8d.png)

geändert?
Und wenn sie dann zurück geht, bleibt das System auch auf Deutsch, aber springt dann irgendwann wieder zurück auf Englisch?
Verstehe ich das richtig?

**Comment by geimist** [comment]
Ja, genau. Interessanterweise steht auch nach der Umstellung der Wert "language.entity.languageValues.en" in der Sprachauswahl. In der Sprachliste ist kein Haken bei irgendeiner Sprache gesetzt. Als Admin kann ich ihre Spracheinstellung auch nicht überschreiben.

Hilft es dir, wenn du die Zugangsdaten mit einem temporären PW zu ihrem Account bekommst?

**Comment by webunraveling** [comment]
Das habe ich beobachtet...

* Ich klicken eine neue Sprache. (e.g. Deutsch) 
* Ich klicken "Back" (auf Deutsch: zurück) beim obere linke Ecke
* Die Sprache für JWManagement hat sich nicht verändert.
* Ich klicken die erfrischen taste.
* Alles gut!

@geimist Hilft das? Entschuldigung, wenn ich nicht richtig verstehe. (Does that help? Sorry if I do not understand correctly). Wenn meine Deutsche ist sehr schrecklich... Sehen :point_down: 

In case that is unreadable. Here it is auf Englisch:

Here is my observation....

* I click a new language.
* I click the back button at the top left.
* The language for JWManagement has not changed.
* I click the refresh button.
* Everything is good!

@MarvinZeising, I've noticed that when I have been testing languages, I must manually refresh the page in my browser for the language to actually take effect. *Maybe* this has something to do with it? Sorry, if I'm missunderstanding geimist. My german is not great and I am using Google Translate to understand some of it.

**Comment by geimist** [comment]
> Here is my observation....
> 
> * I click a new language.
> * I click the back button at the top left.
> * The language for JWManagement has not changed.
no, - Everything is good!
> * I click the refresh button.
now, the language falls back to english

**Comment by webunraveling** [comment]
Thanks for the info @geimist. Sorry you are having this trouble. Hopefully we can find the bug very soon.

**Comment by geimist** [comment]
no trouble 😊
Thank you for your hard work!

**Comment by MarvinZeising** [comment]
> Ja, genau. Ein Reload der Seite bewirkt wieder das Zurückfallen der Spracheinstellung. Interessanterweise steht auch nach der Umstellung der Wert "language.entity.languageValues.en" in der Sprachauswahl ([siehe Screenshot](https://github.com/JWDeveloper/JWManagement/issues/398#issuecomment-508425376)). In der Sprachliste ist kein Haken bei irgendeiner Sprache gesetzt. Als Admin kann ich ihre Spracheinstellung auch nicht überschreiben.
> 
> Hilft es dir, wenn du die Zugangsdaten mit einem temporären PW zu ihrem Account bekommst?

Das hört sich verrückt an. Kann ich eigentlich kaum glauben...
Temporäre Zugangsdaten wären das wirklich sehr praktisch. Die kannst du mir ja an support@jwmanagement.org schicken.

**Comment by MarvinZeising** [comment]
Okay, so the problem really is the username. When updating the language, the user's profile data is validated and in this case, fails the validation because of a duplicate username.
However, the username is not a duplicate, it just contains a dot.
Since the username is put into a regular expression, and dot is a meta character there, the search also finds similar usernames without the dot.
I'll fix this by escaping the username before putting it into the regex.
Thanks for the sign-in data for that account.

**Comment by geimist** [comment]
Oh, da muss man erst einmal darauf kommen …
Ich hoffe, du musstest nicht zu viel Zeit in die Suche investieren!

Vielen Dank, dass du an einem Bugfix arbeitest.
Das Kenntwort kann ich ja nun wieder für die Schwester zurücksetzen, oder brauchst du den Zugang noch?

**Comment by MarvinZeising** [comment]
Klar, gerne.
Ja das Kennwort kannst du wieder zurücksetzen.

**Comment by webunraveling** [comment]
:clap: Gut gemacth. Ich hätte nie gedacht, dorthin zu schauen.

**Comment by geimist** [comment]
Hallo Marvin,
darf ich mal nachfragen, ob du den Bugfix schon implementiert hast? Bitte meine Nachfrage nicht falsch verstehen, aber ich nehme an, dass du es damals gleich in den Dev-Zweig aufgenommen hast, es aber immer noch nicht funktioniert … 🙈
Falls es einfach noch in der Pipeline ist, beachte bitte meine Nachfrage nicht.

Danke
Stephan

**Comment by MarvinZeising** [comment]
Hallo Stephan,
doch, müsste eigentlich schon längst drin sein.
Wenn es immer noch nicht geht, könntest du bitte repro steps anlegen, mit denen es nicht fehlschlägt?
Danke dir

**Comment by geimist** [comment]
> … bitte repro steps anlegen   

Sorry, ich verstehe deine Bitte nicht …

**Comment by MarvinZeising** [comment]
Oh, sorry.
Repro steps (oder Steps to Reproduce) ist eine Schritt für Schritt Liste, die vorgibt, was man nacheinander machen muss um einen Fehler zu verursachen.
Wenn der Fehler also noch einmal auftritt, würde mir so eine Schritt für Schritt Liste helfen den Fehler selbst zu bekommen und ihn dadurch zu verstehen.

**Comment by geimist** [comment]
Ach so, ok. Danke.
Der Fehler besteht ja nach wie vor permanent. Ich schicke dir einfach nochmal Zugangsdaten zu ihrem Account per Mail.

**Comment by geimist** [comment]
Hallo,
der Fehler besteht nach wie vor. Wir haben einige Verkündiger im Projekt, welche einen Punkt im Benutzernamen haben. Zum Teil lassen sich keine Änderungen im Profil vornehmen.
Wurde [dieser Bugfix](https://github.com/JWManagement/JWManagement/issues/398#issuecomment-519903191) inzwischen aufgenommen?
